docs/data/sdk-plugins.md

```diff
@@ -403,4 +403,95 @@ Use a Destination Plugin to send events to a third-party APIs
                 };
               }
             }
-            ```
\ No newline at end of file
+            ```
+
+    ???code-example "Send to Google Tag Manager by pushing events onto the [data layer](https://developers.google.com/tag-platform/tag-manager/web/datalayer) (click to expand)"
+
+        === "JavaScript"
+  
+            ```js
+              import { PluginType } from "@amplitude/analytics-types"
+              export class GTMPlugin {
+                name = "google-tag-manager"
+                type = PluginType.DESTINATION
+
+                constructor(containerId) {
+                  this.containerId = containerId
+                }
+
+                injectGtmScript(containerId) {
+                  if (!window.dataLayer) {
+                    window.dataLayer = window.dataLayer || []
+                    window.dataLayer.push({
+                      "gtm.start": new Date().getTime(),
+                      event: "gtm.js"
+                    })
+                    const head = document.getElementsByTagName("head")[0],
+                      script = document.createElement("script"),
+                      dataLayer = "datalayer" != "dataLayer" ? "&l=" + "datalayer" : ""
+                    script.async = true
+                    script.src =
+                      "https://www.googletagmanager.com/gtm.js?id=" + containerId + dataLayer
+                    head.insertBefore(script, head.firstChild)
+                  }
+                }
+
+                async setup() {
+                  this.injectGtmScript(this.containerId)
+                }
+
+                async execute(event) {
+                  this.injectGtmScript(this.containerId)
```

Comment:
Why the injectGtmScript need to be called twice?

Reply:
It makes more sense to delete the on in `execute()` and leave only the one in `setup` as it puts the GTM installation script to header which is a one-time task